jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifier.scala

```diff
@@ -415,7 +415,9 @@ class XGBoostClassificationModel private[ml](
     var numColsOutput = 0
 
     val rawPredictionUDF = udf { rawPrediction: mutable.WrappedArray[Float] =>
-      Vectors.dense(rawPrediction.map(_.toDouble).toArray)
+      val raw = rawPrediction.map(_.toDouble).toArray
+      val rawPredictions = if (numClasses == 2) Array(-raw(0), raw(0)) else raw
+      Vectors.dense(rawPredictions)
```

Comment:
in XGBoost, rawPrediction means `margin` while in Spark, it means probability, so if you use BinaryClassifierEvaluator, you should pass in the value of `getProbabilityCol` instead of rawPredictionCol

Reply:
I think Spark also distinguishes `margin` and `probability`. For example in GBTClassificationModel the rawPrediction is `margin` and probability is the other thing. Please correct me if I'm wrong.

And if XGBoost has a different name convention, I think xgboost-spark should follow Spark's since it uses Spark's interface.

Reply:
you are right, I cannot track back why I had the wrong but lasting impression that Spark doesn't have margin support... 

you need to fix the unit test though....

Reply:
@yanboliang take a look at this?